File: repo-root/groups/group-01-manage-field/code/managefield/src/main/java/com/jad/efreifive/managefield/dto/CreateReservationRequestDto.java
package com.jad.efreifive.managefield.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CreateReservationRequestDto {

    @NotBlank(message = "status_id is required.")
    @JsonProperty("status_id")
    private String statusId;

    @NotBlank(message = "date is required.")
    private String date;

    @NotBlank(message = "start_time is required.")
    @JsonProperty("start_time")
    private String startTime;

    @NotBlank(message = "end_time is required.")
    @JsonProperty("end_time")
    private String endTime;
}

File: repo-root/groups/group-01-manage-field/code/manage-field/src/main/kotlin/fr/efrei/managefield/controller/dto/request/CreateReservationRequestDto.kt
package fr.efrei.managefield.controller.dto.request

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotBlank

/**
 * HTTP payload accepted when creating a reservation.
 */
data class CreateReservationRequestDto(
    @field:NotBlank
    @JsonProperty("status_id")
    val statusId: String,

    @field:NotBlank
    val date: String,

    @field:NotBlank
    @JsonProperty("start_time")
    val startTime: String,

    @field:NotBlank
    @JsonProperty("end_time")
    val endTime: String
)
